meilisearch/index.py

```diff
@@ -1917,6 +1918,68 @@ def reset_search_cutoff_ms(self) -> TaskInfo:
 
         return TaskInfo(**task)
 
+    # PROXIMITY PRECISION SETTINGS
+
+    def get_proximity_precision(self) -> ProximityPrecision | None:
+        """Get the proximity_precision of the index.
+
+        Returns
+        -------
+        settings:
+            proximity_precision of the index.
+
+        Raises
+        ------
+        MeilisearchApiError
+            An error containing details about why Meilisearch can't process your request. Meilisearch error codes are described here: https://www.meilisearch.com/docs/reference/errors/error_codes#meilisearch-errors
+        """
+        return self.http.get(self.__settings_url_for(self.config.paths.proximity_precision))
```

Comment:
This will return as a camel case string rather than a `ProximityPrecision` enum. For that we will need to convert it.

This also need `from camel_converter import to_snake` at the top of the file.

Additionally the tests will need to be updated to check that the enum is returned correctly rather than the string value.

```suggestion
        response = self.http.get(self.__settings_url_for(self.config.paths.proximity_precision))
        return ProximityPrecision[to_snake(response.json()).upper()]
```